core/templates/components/forms/custom-forms-directives/image-uploader-modal.component.html

```diff
@@ -0,0 +1,221 @@
+          [maxImageSizeInKB]="maxImageSizeInKB">
+        </oppia-image-receiver>
+      </div>
+      <div class="oppia-form-error oppia-form-error-text" *ngIf="invalidImageWarningIsShown">
+        {{ 'I18N_IMAGE_UPLOAD_ERROR' | translate }}
+      </div>
+      <div *ngIf="(isThumbnail() && thumbnailImageDataUrl)">
+        <div [hidden]="!allowedBgColors">
+          <div [hidden]="!(allowedBgColors.length > 1)">Choose background color:</div>
+          <div [hidden]="!(allowedBgColors.length > 1)" class="oppia-thumbnail-colour-select">
+            <div *ngFor="let color of allowedBgColors" [ngStyle]="{'background': color}"
+                 (click)="updateBackgroundColor(color)">
+            </div>
+          </div>
+          <div [hidden]="!(allowedBgColors.length === 1)" class="alert alert-warning">
+            <span>
+              Please check that the {{imageName.toLowerCase()}} below has the following background color:
+              <i class="fa fa-square" [ngStyle]="{'color': allowedBgColors[0]}"></i>.
+              If not, please ensure that you are uploading an image with transparent background. Thanks!
+            </span>
+          </div>
+        </div>
+        <div class="oppia-thumbnail-container e2e-test-thumbnail-container" *ngIf="thumbnailImageDataUrl"
+          [ngStyle]="{'background': previewDescriptionBgColor, 'color': previewDescriptionBgColor ? '#FFFFFF' : '', 'width': aspectRatio === '4:3' ? '248px' : '320px' }">
+          <button class="btn btn-secondary oppia-thumbnail-reset-button e2e-thumbnail-reset-button e2e-test-thumbnail-reset-button"
+                  (click)="reset()">
+            <i class="fas fa-times oppia-vcenter"></i>
+          </button>
+          <div>
+            <oppia-thumbnail-display *ngIf="thumbnailImageDataUrl" [classes]="['oppia-thumbnail-image']"
+              [imgSrc]="thumbnailImageDataUrl" [aspectRatio]="'16:9'" [background]="bgColor">
+            </oppia-thumbnail-display>
+            <div class="oppia-thumbnail-preview-title">
+              {{ previewTitle }}
+            </div>
+            <div class="oppia-thumbnail-preview-description">
+              {{ previewDescription }}
+            </div>
+            <div *ngIf="previewFooter" class="oppia-thumbnail-preview-footer">
+              {{ previewFooter }}
+            </div>
+          </div>
+        </div>
+      </div>
+
+      <div *ngIf="(!isThumbnail() && uploadedImage)">
+        <span>{{ 'I18N_IMAGE_UPLOADER_MODAL_DRAG' | translate }}</span>
+        <div class="oppia-image-uploader-modal-crop-area e2e-test-photo-crop">
+          <button class="btn btn-secondary oppia-image-uploader-modal-reset-button" aria-label="close"
+                  (click)="reset()">
+            <i class="fas fa-times oppia-vcenter"></i>
+          </button>
+          <img id="croppable-image" [src]="uploadedImage" #croppableImage>
+        </div>
+        <div [hidden]="!(allowedBgColors.length > 1)" class="bg-color-text">
```

Comment:
This pattern looks strange. It's better to write without negation - `if length <= 1`. Also, it's repeated too many times, so maybe it's time to extract it to a function as well

Reply:
I created hasMoreThanOneBgColor() to check this. Done.